setup.py

```diff
@@ -1,48 +1,73 @@
 #!/usr/bin/env python
+# Copyright 2008 National Research Foundation (South African Radio Astronomy Observatory)
+# BSD license - see LICENSE for details
+
+from __future__ import absolute_import, division, print_function
+
+import os
+
+from setuptools import find_packages, setup
+
+
+this_directory = os.path.abspath(os.path.dirname(__file__))
+
+files = {"Readme": "README.md", "Changelog": "CHANGELOG.md"}
+
+long_description = ""
+for name, filename in files.items():
+    if not name == 'Readme':
+        long_description += "# {}\n".format(name)
+    with open(os.path.join(this_directory, filename)) as _f:
+        file_contents = _f.read()
+    long_description += file_contents + "\n\n"
 
-from setuptools import setup, find_packages
 
 setup(
     name="katcp",
     description="Karoo Array Telescope Communication Protocol library",
-    author="SKA SA KAT-7 / MeerKAT CAM team",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    author="MeerKAT CAM Team",
     author_email="cam@ska.ac.za",
     packages=find_packages(),
-    scripts=[
-        "scripts/katcp-exampleserver.py",
-        "scripts/katcp-exampleclient.py",
-    ],
-    url='https://github.com/ska-sa/katcp-python',
-    download_url='http://pypi.python.org/pypi/katcp',
+    url="https://github.com/ska-sa/katcp-python",
+    download_url="https://pypi.org/project/katcp/",
     license="BSD",
     classifiers=[
         "Development Status :: 5 - Production/Stable",
         "Intended Audience :: Developers",
         "License :: OSI Approved :: BSD License",
         "Operating System :: OS Independent",
+        "Programming Language :: Python",
         "Programming Language :: Python :: 2",
+        "Programming Language :: Python :: 2.7",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.5",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
         "Topic :: Software Development :: Libraries :: Python Modules",
         "Topic :: Scientific/Engineering :: Astronomy",
     ],
     platforms=["OS Independent"],
+    python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*",
     keywords="kat kat7 ska MeerKAT",
     setup_requires=["katversion"],
     use_katversion=True,
     install_requires=[
         "ply",
-        "tornado>=4.3, <5",
-        "futures",
-        "future"
-    ],
-    # run tests and install these requirements with python setup.py nosetests
-    tests_require=[
-        "unittest2",
-        "nose",
-        "mock",
-        "pylint",
-        "coverage",
-        "nosexcover"
+        "future",
+        "futures; python_version<'3'",
+        "tornado>=4.5.3, <5",
```

Comment:
We don't need to be this strict.  Reverted this to the original `"tornado>=4.3, <5"` - 4.3 does still work with both Python 2 and 3.